Create a new field peh_fl in the database for a Person Experiencing Homelessness

roseeichelmann opened the pull request:
## Associated issues
Closes https://github.com/cityofaustin/atd-data-tech/issues/13561

This issue didn't have specs for what the UI interaction should be, but there really is no other place in the editor for editing the fields of the persons table that makes sense other than the Related Records section, so I put the functionality there.

## Testing
**URL to test:** <!-- VZ URL or Netlify -->
Local!

**Steps to test:**
1. Checkout this branch, start up your local db and run the new migrations in this branch:
```bash
hasura migrate apply --envfile .env.local
hasura metadata apply --envfile .env.local
``` 
3. Load up the VZE pointing to your local db
4. Go to a crash, you should now see the "Experiencing Homelessness" field in the People section of the Related Records table.
5. Test editing the field to be True, False, and back to No Data (null in the db).


---
#### Ship list
- [ ] Check migrations for any conflicts with latest migrations in master branch
- [ ] Code reviewed
- [x] Product manager approved

roseeichelmann commented on atd-vze/src/views/Crashes/personDataMap.js:
```diff
@@ -107,6 +107,12 @@ export const primaryPersonDataMap = [
+      peh_fl: {
+        label: "Experiencing Homelessness",
+        editable: true,
+        format: "boolean",
```
created this new "boolean" format because the "select" format uses lookup tables to determine what choices are in the drop down.. this new column doesn't have a lookup table, its not something we get from CRIS.

frankhereford replied:
No, I think you've made the perfect choice. It's exactly what it is - true/false to the person experiencing homelessness.

roseeichelmann replied:
Hmm thats a good point! I think since it's not in the section with the rest of the "Flags" it doesn't stand out as being abnormal, but I think changing to `YES` and `NO` does make sense from a UI perspective!

roseeichelmann commented on atd-vzd/migrations/default/1699990226463_alter_table_public_atd_txdot_person_add_column_peh_fl/up.sql:
```diff
@@ -0,0 +1,4 @@
+alter table "public"."atd_txdot_person" add column "peh_fl" boolean
+ null;
+
+COMMENT ON COLUMN public.atd_txdot_person.peh_fl IS 'Boolean flag that indicates whether the person was experiencing homelessness at the time of the crash.';
```
i think using postgres comments as much as we can especially for columns is great practice!! ive been noticing we have been taking advantage of this more as a team lately

frankhereford replied:
I need to do much better about this - thank you for mentioning it. I agree with you, it's nothing but good. 🚀

frankhereford reviewed (approved):
I think you've done this perfectly. I've spun it up and watched the records change while editing them in the UI. I think where you put the UI elements make perfect sense. too. This is 🚀🚀🔥.

mddilley reviewed (approved):
Looks great, and I agree that the related records area is a fine place to add this to the UI. 🙏 Testing checks out - thanks for adding this so the VZ team can start capturing this data. 🚢 🚀

johnclary reviewed (commented):
👋 just chiming in to say that this looks good to me! it's cool to look at how y'all manage these re-usable inputs in VZ 🚀

chiaberry reviewed (approved):
I like where you put the field editing and the labels you chose

patrickm02L commented:
Back-channeled with Rose about the UI title "Experiencing Homelessness" and requested the title be changed to "Unhoused" so as to take up less real estate on the People card.

![image](https://github.com/cityofaustin/atd-vz-data/assets/5566727/6afdd58c-5ca1-4bc3-b2e0-87e5906c5a53)

roseeichelmann commented:
I changed the label to say "Unhoused" and also thanks to @mddilley suggestion I changed true/false to be yes/no instead which I think is nicer from a UI standpoint!

patrickm02L reviewed (approved):
Good to go! 💯

mddilley commented:
@roseeichelmann I tested this again locally and looked at the new commits, and it is 🚢✨!
